### PR TITLE
Darius

### PR DESCRIPTION
**Pull request by Lysxia**
Add `do_auto_attack` to automatically apply damage to the target champion. In the future this will also be where procs will be applied.

### CODE REVIEW

File: champion.py [hunk 1]
```diff
@@ -61,6 +62,17 @@ def auto_attack(self, enemy_champion, is_crit: bool = False):
...
+    def take_damage(self, damage):
+        """Takes damage from an enemy champion"""
+
+        self.current_health -= damage
+
+    def do_auto_attack(self, enemy_champion, is_crit: bool = False):
+        """Deals damage to an enemy champion with an autoattack"""
+
+        damage = self.auto_attack_damage(enemy_champion, is_crit)
+        enemy_champion.take_damage(damage)
```
Comment (Lysxia) [inline champion.py:74]: This function was written by copilot.

File: champion.py [hunk 2]
```diff
@@ -23,6 +23,7 @@ def __init__(self, champion_name: str, item_names: Optional[List[str]] = None, l
         self.orig_base_stats = self.get_champion_base_stats(ALL_CHAMPION_BASE_STATS[champion_name])
         self.item_stats = self.inventory.get_items_total_stats(self.inventory.items)
         self.orig_bonus_stats = self.get_bonus_stats()
+        self.current_health = self.orig_base_stats["health"]
```
Comment (PaulQbFeng) [inline champion.py:26]: as we anticipate the buff and debuff that a champion can encounter during a fight, we were thinking of having a base_stats and bonus_stats Stats object that are copies of their `orig` counterpart at the begining of the fight.
We would then alter the health, armor, ad etc inside those dynamic stats objects. 

Or do you think we should have the current stats directly as attribute of the champion

**Comment by PaulQbFeng** [comment]
Thanks sam for helping :))))

File: champion.py (hunk 2 above)
Comment (Lysxia) [reply champion.py:26]: Yeah I think we should just put stats in the champion directly

**Pull request merged**